src/main/scala/za/co/absa/abris/avro/read/confluent/SchemaManagerFactory.scala

```diff
@@ -17,87 +17,55 @@
 
-import scala.collection.JavaConverters._
 import scala.collection.concurrent
+import scala.util.control.NonFatal
 
 /**
  * This thread-safe factory creates [[SchemaManager]] and also manages the instances of SchemaRegistryClient
  * used by allowing caching of the references in order to avoid creating instances in every call that can be
  * used to cache schemas.
- * This factory also allows us to mock the client for testing purposes.
+ * This factory also allows us to use custom registry client via abris.registryClient.class property.
  */
 object SchemaManagerFactory extends Logging {
 
-  private val clientInstances: concurrent.Map[Map[String,String], SchemaRegistryClient] = concurrent.TrieMap()
+  private val clientInstances: concurrent.Map[Map[String,String], AbrisRegistryClient] = concurrent.TrieMap()
 
-  def addSRClientInstance(configs: Map[String, String], client: SchemaRegistryClient): Unit = {
+  @DeveloperApi
+  def addSRClientInstance(configs: Map[String, String], client: AbrisRegistryClient): Unit = {
     clientInstances.put(configs, client)
   }
 
+  @DeveloperApi
   def resetSRClientInstance(): Unit = {
    clientInstances.clear()
   }
 
   def create(configs: Map[String,String]): SchemaManager = new SchemaManager(getOrCreateRegistryClient(configs))
 
-  private def getOrCreateRegistryClient(configs: Map[String,String]): SchemaRegistryClient = {
+  private def getOrCreateRegistryClient(configs: Map[String,String]): AbrisRegistryClient = {
     clientInstances.getOrElseUpdate(configs, {
-      val settings = new KafkaAvroDeserializerConfig(configs.asJava)
-      val urls = settings.getSchemaRegistryUrls
-      val maxSchemaObject = settings.getMaxSchemasPerSubject
-
-      if (hasValidMockURL(urls)) {
-        logInfo(msg = s"Configuring new Schema Registry instance of type " +
-          s"'${classOf[AbrisMockSchemaRegistryClient].getCanonicalName}'")
-
-        new AbrisMockSchemaRegistryClient()
-      } else if (configs.contains(AbrisConfig.REGISTRY_CLIENT_CLASS)) {
-        val cl = Class.forName(configs(AbrisConfig.REGISTRY_CLIENT_CLASS))
-        if (classOf[CustomRegistryClient].isAssignableFrom(cl)) {
-          logInfo(msg = s"Configuring new Schema Registry instance of type " +
-            s"'${cl.getCanonicalName}'")
-          val instance = cl.getDeclaredConstructor(Array(classOf[util.Map[String, String]]): _*)
-            .newInstance(configs.asJava)
-
-          instance.asInstanceOf[CustomRegistryClient]
-        } else {
-          throw new IllegalArgumentException("Custom registry classes have to extend 'CustomRegistryClient'!")
+      if (configs.contains(AbrisConfig.REGISTRY_CLIENT_CLASS)) {
+        try {
+          val clazz = Class.forName(configs(AbrisConfig.REGISTRY_CLIENT_CLASS))
+          logInfo(msg = s"Configuring new Schema Registry client of type '${clazz.getCanonicalName}'")
+          clazz
+            .getDeclaredConstructor(classOf[Map[String, String]])
```

Comment:
Since you expect a constructor that takes a `Map[String, String]`, how about converting `AbrisRegistryClient` to an abstract class, so you can make sure that every class that extends from `AbrisRegistryClient` can be instantiated here?

Reply:
I probably should call a parameterless constructor as fallback. There is no need to force implementation to take the config map if they don't need it.

Reply:
One such case was `MyRegistry.scala` that would need to differ in different confluent versions.
That class was removed.

Reply:
done